### DISCUSSION TITLE
Cannot render email templates without a template engine

### DISCUSSION THREAD

**Discussion by youssef67** [adonisjs/core#4590] [Help] [answered]
Good morning,

I am currently working on version 6 to build API and I am trying to send an email to register. I performed the node ace make:view emails/verify_email command to generate the resources/views/emails folder.

When I run the code in charge of execution 

```ts
import mail from '@adonisjs/mail/services/main'

export default class EmailService {
  static async sendActivationEmail(
    email: string,
    token: string,
    name: string | null,
    urlBase?: string
  ) {
    const activationUrl = `${urlBase}/activate?token=${token}&email=${encodeURIComponent(email)}`

    console.log('class EmailService - sendActivationEmail')

    await mail.send((message) => {
      message
        .from('info@test.com')
        .to(email)
        .subject('Activate Your Account')
        .htmlView('emails/verify_email', { email, activationUrl, name })
    })
  }
}
```

I get an error 

```
/var/www/api-rb/node_modules/@adonisjs/mail/src/message.ts:50
        throw new RuntimeException('Cannot render email templates without a template engine')
              ^


RuntimeException: Cannot render email templates without a template engine
    at getTemplateEngine (/var/www/api-rb/node_modules/@adonisjs/mail/src/message.ts:50:15)
    at computeContentsFor (/var/www/api-rb/node_modules/@adonisjs/mail/src/message.ts:88:28)
    at sendCompiled (/var/www/api-rb/node_modules/@adonisjs/mail/src/mailer.ts:95:19)
    at Mailer.send (/var/www/api-rb/node_modules/@adonisjs/mail/src/mailer.ts:179:17)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at Function.sendActivationEmail (/var/www/api-rb/app/services/email/email_service.ts:14:5)
    at Function.sendActivationEmail (/var/www/api-rb/app/services/activation/activation_service.ts:20:5) {
  status: 500,
  code: 'E_RUNTIME_EXCEPTION'
```

can anyone help me ?

**Comment by Julien-R44** [comment, marked as answer]
You probably forgot to install and configure edge

Docs here: https://docs.adonisjs.com/guides/views-and-templates/edgejs

TLDR : just run `node ace add edge`

**Comment by youssef67** [comment]
thanks for your answer this command but i had an error 

`[ error ] npm install edge.js exited with a status of 1.
          at normalizeResult (/var/www/api-rb/node_modules/@jsdevtools/ez-spawn/lib/normalize-result.js:31:11)
          at ChildProcess.<anonymous> (/var/www/api-rb/node_modules/@jsdevtools/ez-spawn/lib/async.js:79:19)
          at ChildProcess.emit (node:events:519:28)
          at ChildProcess._handle.onexit (node:internal/child_process:294:12)`

**Comment by youssef67** [comment]
I just saw the node ace add edge command in detail and simply added

`() => import('@adonisjs/core/providers/edge_provider')`

in the adonisrc.ts file.

I no longer have the error. However, I have a problem with my Google ID and password.

**Comment by youssef67** [comment]
OK Ok, I had to enable two-factor authentication and create a new password

I found the solution here, I'll let go if necessary
https://stackoverflow.com/questions/72601470/adonisjs-cant-send-email-via-adonisjs-mail

I confirm that I receive the email and this is the first time that I dance with joy upon receiving an email
